src/main/kotlin/no/fdk/dataset/preview/service/PreviewService.kt

```diff
@@ -64,6 +69,45 @@ class PreviewService(
         }
     }
 
+    private fun excelPreview(rows: Int?, body: ResponseBody): Preview {
+        logDebug("Parsing Excel")
+
+        val tableRows = arrayListOf<TableRow>()
+
+        val workbook: Workbook = XSSFWorkbook(body.byteStream())
+        val formatter = DataFormatter()
+        val sheet = workbook.getSheetAt(0)
+
+        var lastCellNum = 0
+        sheet.forEach { row ->
+            val tableRow = TableRow(row.map {
+                formatter.formatCellValue(it)
+            })
+
+            tableRows.add(tableRow)
+
+            lastCellNum = when {
+                row.lastCellNum > lastCellNum &&
```

Comment:
Mulig å dele denne opp litt?
```suggestion
                row.lastCellNum <= lastCellNum -> lastCellNum
                formatter.formatCellValue(row.last()).isNotEmpty() -> row.physicalNumberOfCells
                else -> lastCellNum
```